src/components/DeleteButton/DeleteButton.tsx

```diff
@@ -0,0 +1,121 @@
+type DeleteButtonProps = {
+  contentType: DeleteButtonContentType;
+  handleDelete: () => void;
+  error: string | null;
+  setError: (error: string | null) => void;
+  plants: number;
+};
+
+export default function DeleteButton({
+  contentType,
+  handleDelete,
+  error,
+  setError,
+  plants,
+}: DeleteButtonProps) {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [disclaimer, setDisclaimer] = useState("");
+
+  const toggleModal = () => {
+    setModalOpen(!modalOpen);
+    setError(null);
+  };
+
+  const dislaimerMessage = () => {
+    setDisclaimer("Room with plants cannot be deleted");
+  };
+  // delete room
+
+  if (contentType === DeleteButtonContentType.ROOM) {
+    return (
+      <>
+        <StyledDeleteButton onClick={toggleModal}>
+          <RiDeleteBin2Fill size={25} />
+        </StyledDeleteButton>
+
+        {modalOpen && (
+          <>
+            <StyledOverlay onClick={toggleModal} />
+            <StyledModal>
+              <StyledModalContent>
+                <StyledModalHeadline>
+                  Do you really want to delete this {contentType}?
+                </StyledModalHeadline>
+                <br />
+                <StyledConfirmButton type="submit" onClick={toggleModal}>
+                  No, go back!
+                </StyledConfirmButton>
+                <StyledConfirmButton
+                  type="submit"
+                  onClick={plants ? dislaimerMessage : handleDelete}
+                >
+                  Yes, I am sure!
+                </StyledConfirmButton>
+                {disclaimer}
+                <StyledCloseModalBtn onClick={toggleModal}>
+                  ✕
+                </StyledCloseModalBtn>
+              </StyledModalContent>
+            </StyledModal>
+          </>
+        )}
+      </>
+    );
+  }
+
+  // delete plant
+
+  if (contentType === DeleteButtonContentType.PLANT) {
+    return (
+      <>
+        <StyledDeleteButton onClick={toggleModal}>
+          <RiDeleteBin2Fill size={25} />
+        </StyledDeleteButton>
+
+        {modalOpen && (
+          <>
+            <StyledOverlay onClick={toggleModal} />
+            <StyledModal>
+              <StyledModalContent>
+                {error && <div>Error: {error}</div>}
+
+                <StyledModalHeadline>
+                  Do you really want to delete this {contentType}?
+                </StyledModalHeadline>
+                <br />
+                <StyledConfirmButton type="submit" onClick={toggleModal}>
+                  No, go back!
+                </StyledConfirmButton>
+                <StyledConfirmButton type="submit" onClick={handleDelete}>
+                  Yes, I am sure!
+                </StyledConfirmButton>
+
+                <StyledCloseModalBtn onClick={toggleModal}>
+                  ✕
+                </StyledCloseModalBtn>
+              </StyledModalContent>
+            </StyledModal>
+          </>
+        )}
+      </>
+    );
+  }
+}
```

Comment:
You could potentially add a default case which throws/shows an error, it shouldn't be possible to have this error state but never say never (as you have two `if` blocks) here, e.g.:

```suggestion
  }
  
  throw new Error("Unexpected type for delete button");
}
```

Reply:
```suggestion
  } else {
    throw new Error("Unexpected type for delete button");
  }
```

Do i need the else or can i simply add the throw new Error(..) after the bracket?